File: Fundamentals/js/todo_one.js
// Setting and Swapping

// Set myNumber to 42. Set myName to your name. Now swap myNumber into myName & vice versa. 

// these will be my global var

function set_swap(myNumber, myName ) {
    let temp = myNumber;
    myNumber = myName;
    myName = temp;
    console.log('my name is ' + myName)
    console.log('my number is ' + myNumber)
}


// swapped the name and number but I do have my name and the number but 'undefined' at the end of the run
// this was a successful swap so far
// console.log(set_swap('Dalius', 42 ))

// ------------------------------------------------------------------------------------------------------

// Print -52 to 1066

// Print integers from -52 to 1066 using a FOR loop. 

function loopty_loop() {
    for (let i = -52; i <= 1066; i++) {
        console.log(i);
    }
}

// console.log(loopty_loop());

// -------------------------------------------------------------------------------------


// Don’t Worry, Be Happy

// Create beCheerful(). Within it, console.log string "good morning!" Call it 98 times.

// note: this is a for loop if we are going to call this 98 times in the console
// I think we would have to do the .length() function for this one or maybe not...
function beCheerful() {
    for (let i = 1; i <= 98; i++) {
        console.log("good morning!");
    }
}
// console.log(beCheerful());

// ------------------------------------------------------------------------

// Multiples of Three – but Not All

// Using FOR, print multiples of 3 from -300 to 0. Skip -3 and -6.

function multiple_of_three() {
    for (let i = -300; i <= 0; i += 3) {
        if (i === -3 || i === -6) {
          continue; // Skip -3 and -6
        }
        console.log(i);
    }
}
// console.log(multiple_of_three());

// ---------------------------------------------------------------------

// Printing Integers with While

// Print integers from 2000 to 5280, using a WHILE.

// explanation: In this code, we initialize the variable num to 2000,
// and then we use a while loop to repeatedly execute the block of code inside it until 
// the condition num <= 5280 becomes false. Inside the loop, 
// we print the value of num to the console, and then we increment its value by 1 using the num++ statement.

// The loop will keep running as long as num is less than or equal to 5280,
// so it will print all integers from 2000 to 5280, inclusive.

function while_int(x,y) {
    let num = x;

    while (num <= y) {
        console.log(num);
        num++;
    }

} 
console.log(while_int(2000, 5280))
// ---------------------------------------------------------------------

File: Fundamentals/python/todo_one.py
# Print -52 to 1066

# Print integers from -52 to 1066 using a FOR loop. 

# in python we would use range() for our loops in python
# range(start, stop, step)
def loopty():
    for i in range(-52, 1067):
        print(i)
# print(loopty())

# Don’t Worry, Be Happy

# Create beCheerful(). Within it, console.log string "good morning!" Call it 98 times.

def beCheerful():
    for i in range(1,98):
        print("good morning!")
# print(beCheerful())

# Multiples of Three – but Not All

# Using FOR, print multiples of 3 from -300 to 0. Skip -3 and -6.

# explanation: By setting range(-300, 1, 3), we create a sequence that starts from -300, goes up to 0
# (exclusive), and increments by 3 at each step. The if statement inside the loop is used to skip printing -3 and -6.
# we are skipping by 3's but aren't looking through a sequence of 3

def multiples_of_three():
    for i in range (-300, 1, 3):
        if i == -3 or i == -6:
            continue
        print(i)
# print(multiples_of_three())

# Printing Integers with While

# Print integers from 2000 to 5280, using a WHILE.

# explanation: In this code, we start with num initialized to 2000, and then we use a while loop to keep
# printing num as long as it is less than or equal to 5280. After each iteration, we increment num by 1 using
# num += 1 to move to the next integer in the sequence. The loop will stop once num becomes greater than 5280.

# I had to re-teach myself how to use a while loop

# def while_int(x,y):
#     num = x

# while num <= y:
#     print(num)
#     num += 1

# print(while_int(2000, 5280))

#  ----------------------------------------

# Leap Year

# Write a function that determines whether a given year is a leap year. If a year is divisible by four, it is a leap year, unless it is divisible by 100. However, if it is divisible by 400, then it is.

# explanation: did this one in first year of college...

def is_leap_year(year):
    if year % 4 == 0:
        if year % 100 == 0:
            if year % 400 == 0:
                return True
            else:
                return False
        else:
            return True
    else:
        return False

# Test cases
# print(is_leap_year(2000))  # True
# print(is_leap_year(2020))  # True
# print(is_leap_year(1900))  # False
# print(is_leap_year(2023))  # False

#  ------------------------------------------

# Print and Count
# Print all integer multiples of 5, from 512 to 4096. Afterward, also log how many there were.

# count = 0

# for i in range(512, 4097):
#     if i % 5 == 0:
#         print(i)
#         count += 1

# print("Total count:", count)

# Multiples of Six

# Print multiples of 6 up to 60,000, using a WHILE.

def mult_of_six():
    pass